src/Hub/Transport/Redis/RedisTransport.php

```diff
@@ -41,8 +44,24 @@ public function __construct(
             'trimInterval' => 0.0,
             'channel' => 'mercure',
             'key' => 'mercureUpdates',
+            'pingInterval' => 2.0,
         ]);
         $this->options = $resolver->resolve($options);
+        if ($this->options['pingInterval']) {
+            Loop::addPeriodicTimer($this->options['pingInterval'], fn () => $this->ping());
+        }
+    }
+
+    /**
+     * @codeCoverageIgnore
+     */
+    private function ping(): void
+    {
+        try {
+            await($this->redis->ping()); // @phpstan-ignore-line
```

Comment:
I think it is an error to use `await` for the call of ping method of Redis. It is a synchronous response, right? In local tests, I had 502 from Caddy for `/.well-known/mercure?topic=*`. Without the `await`, it seems better.

Reply:
But it doesn't work anyway for mercure subscription.

Reply:
In fact, it is strange because `_call` of `Clue\React\Redis\Client` returns a `PromiseInterface`.

Reply:
It seems the promise is not always resolved or rejected. In https://github.com/clue/reactphp-redis/blob/fccbda7b168306a6acfe491059a58454464f2f42/src/Io/StreamingClient.php#L136, it emits an event without resolving or rejecting the promise.

With `await`, it seems to wait forever.

Reply:
Seems to work better with the traditional `then()` approach - see [ca716bf](https://github.com/bpolaszek/freddie/pull/25/commits/ca716bf3527a8a66ae5120e1ee3e4d5e37d02417)